src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -396,8 +396,10 @@ public async Task ReadResultToEnd(DbDataReader dbDataReader, CancellationToken c
                     //
                     availableTask = SendCurrentResults();
 
-                    while (await dataReader.ReadAsync(cancellationToken))
+                    while (dataReader.Read())
                     {
+                        // check for cancellation token before actually making connection
```

Comment:
This doesn't seem accurate as the connection has already been made...and also same comment about this being self-documenting.

Reply:
I probably just copied over comments in excitement 😄